app/src/main/java/net/squanchy/tweets/service/TwitterRepository.kt

```diff
@@ -0,0 +1,28 @@
+package net.squanchy.tweets.service
+
+import com.twitter.sdk.android.core.TwitterCore
+import com.twitter.sdk.android.core.models.Search
+import com.twitter.sdk.android.core.services.SearchService
+
+import io.reactivex.Single
+import io.reactivex.schedulers.Schedulers
+import retrofit2.Call
+
+class TwitterRepository {
+
+    private val searchService = TwitterCore.getInstance().apiClient.searchService
+
+    internal fun load(query: String): Single<Search> {
+        return Single.fromCallable { createSearchRequest(query).execute().body() }
+            .subscribeOn(Schedulers.io())
+    }
+
+    @Throws(Exception::class)
```

Comment:
Do we need this? Couldn't this at least be more strictly typed?

Reply:
Definitely